From 06ec60c67bfacb8466eaa03824ad5611bd9ba322 Mon Sep 17 00:00:00 2001
From: Anson Huang <Anson.Huang@nxp.com>
Date: Mon, 15 May 2017 17:08:58 +0800
Subject: [PATCH 1781/5242] MLK-14902 ARM64: dts: freescale: imx8qm: make
 default dtb boot up all cores

commit  be7f495f08e7d76d36f766256677b3eea1e54863 from
https://source.codeaurora.org/external/imx/linux-imx.git

Change dtb's core settings as below on i.MX8QM:

boot up 4 CA53 and 2 CA72 (using CA53 specific flash.bin)
	fsl-imx8qm-lpddr4-arm2.dts

boot up 4 CA53 (using CA53 specific flash.bin)
	fsl-imx8qm-lpddr4-arm2_ca53.dts

boot up 2 CA72 (using CA72 specific flash.bin)
	fsl-imx8qm-lpddr4-arm2_ca72.dts

Signed-off-by: Anson Huang <Anson.Huang@nxp.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 arch/arm64/boot/dts/freescale/Makefile             |    2 +-
 .../boot/dts/freescale/fsl-imx8qm-lpddr4-arm2.dts  |    8 +++++
 .../dts/freescale/fsl-imx8qm-lpddr4-arm2_ca53.dts  |   24 +++++++++++++++
 .../dts/freescale/fsl-imx8qm-lpddr4-arm2_ca72.dts  |   21 ++++++++-----
 .../freescale/fsl-imx8qm-lpddr4-arm2_ca72_only.dts |   32 --------------------
 arch/arm64/boot/dts/freescale/fsl-imx8qm.dtsi      |    1 +
 6 files changed, 48 insertions(+), 40 deletions(-)
 create mode 100644 arch/arm64/boot/dts/freescale/fsl-imx8qm-lpddr4-arm2_ca53.dts
 delete mode 100644 arch/arm64/boot/dts/freescale/fsl-imx8qm-lpddr4-arm2_ca72_only.dts

diff --git a/arch/arm64/boot/dts/freescale/Makefile b/arch/arm64/boot/dts/freescale/Makefile
index 2d196e6..7915481 100644
--- a/arch/arm64/boot/dts/freescale/Makefile
+++ b/arch/arm64/boot/dts/freescale/Makefile
@@ -13,4 +13,4 @@ dtb-$(CONFIG_ARCH_LAYERSCAPE) += fsl-ls2080a-rdb.dtb
 dtb-$(CONFIG_ARCH_LAYERSCAPE) += fsl-ls2080a-simu.dtb
 dtb-$(CONFIG_ARCH_LAYERSCAPE) += fsl-ls2088a-qds.dtb
 dtb-$(CONFIG_ARCH_LAYERSCAPE) += fsl-ls2088a-rdb.dtb
-dtb-$(CONFIG_ARCH_FSL_IMX8QM) += fsl-imx8qm-lpddr4-arm2.dtb fsl-imx8qm-lpddr4-arm2_ca72.dtb fsl-imx8qm-lpddr4-arm2_ca72_only.dtb
+dtb-$(CONFIG_ARCH_FSL_IMX8QM) += fsl-imx8qm-lpddr4-arm2.dtb fsl-imx8qm-lpddr4-arm2_ca53.dtb fsl-imx8qm-lpddr4-arm2_ca72.dtb
diff --git a/arch/arm64/boot/dts/freescale/fsl-imx8qm-lpddr4-arm2.dts b/arch/arm64/boot/dts/freescale/fsl-imx8qm-lpddr4-arm2.dts
index d4dd005..2412568 100644
--- a/arch/arm64/boot/dts/freescale/fsl-imx8qm-lpddr4-arm2.dts
+++ b/arch/arm64/boot/dts/freescale/fsl-imx8qm-lpddr4-arm2.dts
@@ -330,3 +330,11 @@
 	>;
 	clocks = <&clk IMX8QM_A53_DIV>;
 };
+
+&A72_0 {
+	operating-points = <
+		/* kHz    uV */
+		1800000 1150000
+	>;
+	clocks = <&clk IMX8QM_A72_DIV>;
+};
diff --git a/arch/arm64/boot/dts/freescale/fsl-imx8qm-lpddr4-arm2_ca53.dts b/arch/arm64/boot/dts/freescale/fsl-imx8qm-lpddr4-arm2_ca53.dts
new file mode 100644
index 0000000..9c42d98
--- /dev/null
+++ b/arch/arm64/boot/dts/freescale/fsl-imx8qm-lpddr4-arm2_ca53.dts
@@ -0,0 +1,24 @@
+/*
+ * Copyright (C) 2016 Freescale Semiconductor, Inc.
+ * Copyright 2017 NXP
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public License
+ * as published by the Free Software Foundation; either version 2
+ * of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ */
+
+#include "fsl-imx8qm-lpddr4-arm2.dts"
+
+&A72_0 {
+	device_type = "";
+};
+
+&A72_1 {
+	device_type = "";
+};
diff --git a/arch/arm64/boot/dts/freescale/fsl-imx8qm-lpddr4-arm2_ca72.dts b/arch/arm64/boot/dts/freescale/fsl-imx8qm-lpddr4-arm2_ca72.dts
index 7eaf12d..55e2e8c 100644
--- a/arch/arm64/boot/dts/freescale/fsl-imx8qm-lpddr4-arm2_ca72.dts
+++ b/arch/arm64/boot/dts/freescale/fsl-imx8qm-lpddr4-arm2_ca72.dts
@@ -14,12 +14,19 @@
  */
 
 #include "fsl-imx8qm-lpddr4-arm2.dts"
-#include "fsl-imx8-ca72.dtsi"
 
-&A72_0 {
-	operating-points = <
-		/* kHz    uV */
-		1800000 1150000
-	>;
-	clocks = <&clk IMX8QM_A72_DIV>;
+&A53_0 {
+	device_type = "";
+};
+
+&A53_1 {
+	device_type = "";
+};
+
+&A53_2 {
+	device_type = "";
+};
+
+&A53_3 {
+	device_type = "";
 };
diff --git a/arch/arm64/boot/dts/freescale/fsl-imx8qm-lpddr4-arm2_ca72_only.dts b/arch/arm64/boot/dts/freescale/fsl-imx8qm-lpddr4-arm2_ca72_only.dts
deleted file mode 100644
index a3011d2..0000000
--- a/arch/arm64/boot/dts/freescale/fsl-imx8qm-lpddr4-arm2_ca72_only.dts
+++ /dev/null
@@ -1,32 +0,0 @@
-/*
- * Copyright (C) 2016 Freescale Semiconductor, Inc.
- * Copyright 2017 NXP
- *
- * This program is free software; you can redistribute it and/or
- * modify it under the terms of the GNU General Public License
- * as published by the Free Software Foundation; either version 2
- * of the License, or (at your option) any later version.
- *
- * This program is distributed in the hope that it will be useful,
- * but WITHOUT ANY WARRANTY; without even the implied warranty of
- * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
- * GNU General Public License for more details.
- */
-
-#include "fsl-imx8qm-lpddr4-arm2_ca72.dts"
-
-&A53_0 {
-	device_type = "";
-};
-
-&A53_1 {
-	device_type = "";
-};
-
-&A53_2 {
-	device_type = "";
-};
-
-&A53_3 {
-	device_type = "";
-};
diff --git a/arch/arm64/boot/dts/freescale/fsl-imx8qm.dtsi b/arch/arm64/boot/dts/freescale/fsl-imx8qm.dtsi
index a4abd2c..97253f3 100644
--- a/arch/arm64/boot/dts/freescale/fsl-imx8qm.dtsi
+++ b/arch/arm64/boot/dts/freescale/fsl-imx8qm.dtsi
@@ -15,6 +15,7 @@
 
 #include <dt-bindings/interrupt-controller/arm-gic.h>
 #include "fsl-imx8-ca53.dtsi"
+#include "fsl-imx8-ca72.dtsi"
 #include <dt-bindings/clock/imx8qm-clock.h>
 #include <dt-bindings/soc/imx_rsrc.h>
 #include <dt-bindings/soc/imx8_pd.h>
-- 
1.7.9.5

